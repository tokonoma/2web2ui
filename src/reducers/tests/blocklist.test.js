import cases from 'jest-in-case';
import blocklistReducer, { initialState } from '../blocklist';

const TEST_CASES = {
  'list incidents pending': {
    type: 'LIST_INCIDENTS_PENDING',
  },
  'list incidents success': {
    payload: { fakeData: true },
    type: 'LIST_INCIDENTS_SUCCESS',
  },
  'list incidents fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_INCIDENTS_FAIL',
  },
  'get incident pending': {
    type: 'GET_INCIDENT_PENDING',
  },
  'get incident success': {
    payload: { fakeData: true },
    type: 'GET_INCIDENT_SUCCESS',
  },
  'get incident fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'GET_INCIDENT_FAIL',
  },
  'list incidents for resource pending': {
    type: 'LIST_INCIDENTS_FOR_RESOURCE_PENDING',
  },
  'list incidents for resource success': {
    payload: { fakeData: true },
    type: 'LIST_INCIDENTS_FOR_RESOURCE_SUCCESS',
  },
  'list incidents for resource fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_INCIDENTS_FOR_RESOURCE_FAIL',
  },
  'list incidents for blocklist pending': {
    type: 'LIST_INCIDENTS_FOR_BLOCKLIST_PENDING',
  },
  'list incidents for blocklist success': {
    payload: { fakeData: true },
    type: 'LIST_INCIDENTS_FOR_BLOCKLIST_SUCCESS',
  },
  'list incidents for blocklist fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_INCIDENTS_FOR_BLOCKLIST_FAIL',
  },
  'list historical incidents pending': {
    type: 'LIST_HISTORICAL_INCIDENTS_PENDING',
  },
  'list historical incidents success': {
    payload: { fakeData: true },
    type: 'LIST_HISTORICAL_INCIDENTS_SUCCESS',
  },
  'list historical incidents fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_HISTORICAL_INCIDENTS_FAIL',
  },
  'list monitors pending': {
    type: 'LIST_MONITORS_PENDING',
  },
  'list monitors success': {
    payload: { fakeData: true },
    type: 'LIST_MONITORS_SUCCESS',
  },
  'list monitors fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_MONITORS_FAIL',
  },
  'add watchlist pending': {
    type: 'ADD_WATCHLIST_PENDING',
  },
  'add watchlist success': {
    payload: { fakeData: true },
    type: 'ADD_WATCHLIST_SUCCESS',
  },
  'add watchlist fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'ADD_WATCHLIST_FAIL',
  },
  'delete monitor pending': {
    type: 'DELETE_MONITOR_PENDING',
  },
  'delete monitor fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'DELETE_MONITOR_FAIL',
  },
  'list blocklists pending': {
    type: 'LIST_BLOCKLISTS_PENDING',
  },
  'list blocklists success': {
    payload: { fakeData: true },
    type: 'LIST_BLOCKLISTS_SUCCESS',
  },
  'list blocklists fail': {
    payload: { errors: [{ message: 'Some error occurred' }] },
    type: 'LIST_BLOCKLISTS_FAIL',
  },
};

cases(
  'BlockList Reducer',
  action => {
    expect(blocklistReducer(initialState, action)).toMatchSnapshot();
  },
  TEST_CASES,
);

it('BlockList Reducer delete monitor success deletes the resource from redux store list', () => {
  const state = { ...initialState, monitors: [{ resource: '101.101' }, { resource: '101.102' }] };
  const action = {
    type: 'DELETE_MONITOR_SUCCESS',
    meta: {
      resource: '101.101',
    },
  };
  expect(blocklistReducer(state, action).monitors).toEqual([{ resource: '101.102' }]);
});
