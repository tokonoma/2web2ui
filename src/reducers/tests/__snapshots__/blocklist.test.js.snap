// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockList Reducer add watchlist fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
  "watchlistAddError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "watchlistAddPending": false,
}
`;

exports[`BlockList Reducer add watchlist pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
  "watchlistAddError": null,
  "watchlistAddPending": true,
}
`;

exports[`BlockList Reducer add watchlist success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
  "watchlistAddError": null,
  "watchlistAddPending": false,
}
`;

exports[`BlockList Reducer delete monitor fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer delete monitor pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": true,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer get incident fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidentError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "incidentPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer get incident pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidentError": null,
  "incidentPending": true,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer get incident success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incident": Object {
    "fakeData": true,
  },
  "incidentError": null,
  "incidentPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list blocklists fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list blocklists pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": true,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list blocklists success 1`] = `
Object {
  "blocklists": Object {
    "fakeData": true,
  },
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list historical incidents fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "historicalIncidentsError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "historicalIncidentsPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list historical incidents pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "historicalIncidentsError": null,
  "historicalIncidentsPending": true,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list historical incidents success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "historicalIncidents": Object {
    "fakeData": true,
  },
  "historicalIncidentsError": null,
  "historicalIncidentsPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for blocklist fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForBlocklistError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "incidentsForBlocklistPending": false,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for blocklist pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForBlocklistError": null,
  "incidentsForBlocklistPending": true,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for blocklist success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForBlocklist": Object {
    "fakeData": true,
  },
  "incidentsForBlocklistError": null,
  "incidentsForBlocklistPending": false,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for resource fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForResourceError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "incidentsForResourcePending": false,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for resource pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForResourceError": null,
  "incidentsForResourcePending": true,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents for resource success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsForResource": Object {
    "fakeData": true,
  },
  "incidentsForResourceError": null,
  "incidentsForResourcePending": false,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": true,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list incidents success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Object {
    "fakeData": true,
  },
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list monitors fail 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "monitorsPending": false,
}
`;

exports[`BlockList Reducer list monitors pending 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Array [],
  "monitorsError": null,
  "monitorsPending": true,
}
`;

exports[`BlockList Reducer list monitors success 1`] = `
Object {
  "blocklists": Array [],
  "blocklistsError": null,
  "blocklistsPending": false,
  "deleteMonitorError": null,
  "deleteMonitorPending": false,
  "incidents": Array [],
  "incidentsError": null,
  "incidentsPending": false,
  "monitors": Object {
    "fakeData": true,
  },
  "monitorsError": null,
  "monitorsPending": false,
}
`;
